Dashboard 500: DashboardSummary not Serializable, Redis cache.put fails

## Problem

Hitting \`/dashboard\` as an authenticated user returns 500 when Redis caching is active. Root cause from the app log:

\`\`\`
Caused by: java.lang.IllegalArgumentException: DefaultSerializer requires a Serializable payload
    but received an object of type [com.majordomo.domain.model.DashboardSummary]
    at org.springframework.core.serializer.DefaultSerializer.serialize(DefaultSerializer.java:43)
    at org.springframework.data.redis.cache.RedisCache.put(RedisCache.java:204)
    at com.majordomo.application.DashboardService\$\$SpringCGLIB\$\$0.getSummary(<generated>)
    at com.majordomo.adapter.in.web.DashboardPageController.dashboard(DashboardPageController.java:53)
\`\`\`

\`DashboardService.getSummary(...)\` is \`@Cacheable\` (via \`majordomo:\` Redis key prefix). On every first request per org, Spring calls \`RedisCache.put(...)\`, which tries the default JDK \`DefaultSerializer\` — but \`DashboardSummary\` is a plain record that doesn't implement \`Serializable\`, so serialization fails and the exception propagates as a 500 to the user.

This is pre-existing on \`main\` — reproduce with \`docker compose up -d db redis && ./mvnw spring-boot:run\`, log in, navigate to \`/dashboard\`.

## Fix options

1. **Preferred — switch Redis cache serializer to JSON.** Configure a \`RedisCacheConfiguration\` bean with \`GenericJackson2JsonRedisSerializer\` (or equivalent) instead of the JDK default. JSON values are more portable, human-readable in \`redis-cli\`, and don't require every cached type to implement \`Serializable\`.
2. **Alternative — make \`DashboardSummary\` and every nested type (\`MaintenanceSchedule\`, \`ServiceRecord\`, …) implement \`Serializable\`.** Works but imposes a marker interface on domain records and is fragile as the model grows.
3. **Quick workaround while a fix lands:** set \`SPRING_CACHE_TYPE=none\` in your shell / \`docker-compose.override.yml\` to disable the cache locally.

## Acceptance

- [ ] Fresh login → \`/dashboard\` renders (even with empty counts)
- [ ] A second \`/dashboard\` request within the 5-minute TTL hits the cache (verify via \`redis-cli KEYS 'majordomo:*'\`)
- [ ] No \`SerializationException\` in the app log

--- a/src/main/java/com/majordomo/adapter/in/web/config/CacheConfig.java
+++ b/src/main/java/com/majordomo/adapter/in/web/config/CacheConfig.java
@@ -1,0 +1,63 @@
+package com.majordomo.adapter.in.web.config;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.redis.cache.RedisCacheConfiguration;
+import org.springframework.data.redis.serializer.GenericJackson2JsonRedisSerializer;
+import org.springframework.data.redis.serializer.RedisSerializationContext;
+import org.springframework.data.redis.serializer.StringRedisSerializer;
+
+import java.time.Duration;
+
+/**
+ * Redis cache configuration. Replaces the default JDK serializer (which requires
+ * cached value types to implement {@code Serializable}) with a Jackson-based
+ * JSON serializer, so plain records like {@code DashboardSummary} cache without
+ * needing to declare {@code Serializable} on every type in the value graph.
+ *
+ * <p>Defaults pulled from {@code spring.cache.redis.*} properties (TTL, key prefix).</p>
+ */
+@Configuration
+public class CacheConfig {
+
+    /**
+     * Builds the {@link RedisCacheConfiguration} bean used by Spring Boot's
+     * auto-configured {@code RedisCacheManager}. Setting this bean overrides
+     * the default JDK serialization without disabling auto-configuration.
+     *
+     * @param ttlMillis  cache entry TTL in milliseconds (from
+     *                   {@code spring.cache.redis.time-to-live})
+     * @param keyPrefix  prefix prepended to every cache key (from
+     *                   {@code spring.cache.redis.key-prefix})
+     * @return a {@link RedisCacheConfiguration} with JSON value serialization
+     */
+    @Bean
+    public RedisCacheConfiguration redisCacheConfiguration(
+            @Value("${spring.cache.redis.time-to-live:300000}") long ttlMillis,
+            @Value("${spring.cache.redis.key-prefix:}") String keyPrefix) {
+        ObjectMapper mapper = JsonMapper.builder()
+                .addModule(new JavaTimeModule())
+                .addModule(new Jdk8Module())
+                .build();
+        mapper.activateDefaultTyping(
+                mapper.getPolymorphicTypeValidator(),
+                ObjectMapper.DefaultTyping.NON_FINAL);
+        var jsonSerializer = new GenericJackson2JsonRedisSerializer(mapper);
+
+        RedisCacheConfiguration config = RedisCacheConfiguration.defaultCacheConfig()
+                .entryTtl(Duration.ofMillis(ttlMillis))
+                .serializeKeysWith(RedisSerializationContext.SerializationPair
+                        .fromSerializer(new StringRedisSerializer()))
+                .serializeValuesWith(RedisSerializationContext.SerializationPair
+                        .fromSerializer(jsonSerializer));
+        if (keyPrefix != null && !keyPrefix.isBlank()) {
+            config = config.computePrefixWith(name -> keyPrefix + name + "::");
+        }
+        return config;
+    }
+}

--- a/src/test/java/com/majordomo/adapter/in/web/config/CacheConfigTest.java
+++ b/src/test/java/com/majordomo/adapter/in/web/config/CacheConfigTest.java
@@ -1,0 +1,48 @@
+package com.majordomo.adapter.in.web.config;
+
+import com.majordomo.domain.model.DashboardSummary;
+import org.junit.jupiter.api.Test;
+import org.springframework.data.redis.cache.RedisCacheConfiguration;
+
+import java.math.BigDecimal;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Verifies the Redis cache uses JSON serialization, so plain records without
+ * {@code implements Serializable} round-trip cleanly. Regression test for #125.
+ */
+class CacheConfigTest {
+
+    private final CacheConfig config = new CacheConfig();
+
+    @Test
+    void serializesNonSerializableRecord() {
+        RedisCacheConfiguration cacheConfig =
+                config.redisCacheConfiguration(300000L, "majordomo:");
+
+        DashboardSummary summary = new DashboardSummary(
+                3, 7, List.of(), List.of(), List.of(), new BigDecimal("123.45"));
+
+        ByteBuffer buf = cacheConfig.getValueSerializationPair().write(summary);
+        byte[] bytes = new byte[buf.remaining()];
+        buf.get(bytes);
+
+        assertThat(bytes).isNotEmpty();
+        String json = new String(bytes, StandardCharsets.UTF_8);
+        assertThat(json).contains("\"propertyCount\"").contains("123.45");
+    }
+
+    @Test
+    void appliesKeyPrefix() {
+        RedisCacheConfiguration cacheConfig =
+                config.redisCacheConfiguration(300000L, "majordomo:");
+
+        String fullKey = cacheConfig.getKeyPrefixFor("dashboard");
+
+        assertThat(fullKey).startsWith("majordomo:").contains("dashboard");
+    }
+}

